gradio/routes.py

```diff
@@ -210,8 +211,15 @@ def api_docs(request: Request):
 @app.post("/api/predict/", dependencies=[Depends(login_check)])
 async def predict(request: Request, username: str = Depends(get_current_user)):
     body = await request.json()
+    session_hash = body.get("session_hash", None)  # TODO(aliabid94): send from frontend
```

Comment:
@aliabid94 can you have the frontend send a hash back here?

Reply:
@abidlabs instead of getting smt from the request body it is better to use validations in FastAPI.

Reply:
Thanks for the tip @FarukOzderim. I've updated to use pydantic models for this route.

There are a few other routes were we should make this change. Opened #777 to address this